### PR TITLE
Add temporary protected fetch solution by leveraging web server's HTTP Basic Auth

### PR DESCRIPTION
**Pull request by taoza**
As discussed here: https://github.com/bundler/gemstash/issues/24#issuecomment-195382240

/cc @smellsblue

### CODE REVIEW

File: docs/private_gems.md [hunk 1]
```diff
@@ -138,3 +138,35 @@ Gemstash you are interacting with private gems. Gemstash will only let you
 unyank from `/private`. Unlike pushing, Rubygems doesn't support `--host` for
 unyank and yank (yet), so you need to specify the host via the `RUBYGEMS_HOST`
 environment variable.
+
+## Protected Fetching
+
+Private gems often require protected fetching. While the feature is still being discussed at here: https://github.com/bundler/gemstash/issues/24, a temporary solution is available through configuring web server.
+
+Depends on your choice of the web server, for example, Nginx has a `basic_auth` module, which helps to setup HTTP Basic Authentication. On the Bundler side, HTTP Basic Auth credentials can be configured through: http://bundler.io/man/gemfile.5.html#CREDENTIALS-credentials
+
```
Comment (pcarranza) [inline docs/private_gems.md]: Could you also provide a sample on how to setup a basic auth? (rather than pointing to the manual)

Comment (taoza) [inline docs/private_gems.md]: @pcarranza Sure thing. Please let me know if this helps: https://github.com/taoza/gemstash/commit/c599aab86034507e7eb91c51ab4e515a790cd657

**Comment by taoza** [comment]
@pcarranza @smellsblue How do you think about the new doc section?

**Comment by smellsblue** [comment]
Looks good to me! Thanks for the work!

**Pull request merged**